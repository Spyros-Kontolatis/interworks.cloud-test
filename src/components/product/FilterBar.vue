<template>
  <div class="flex flex-col md:flex-row w-full justify-between items-center">
    <div
      v-if="sortingOptions?.length"
      class="flex gap-4 items-center w-full md:w-auto"
    >
      <h5 class="font-extrabold text-start">Sort:</h5>
      <select
        class="mt-2 p-4 bg-website-background border border-primary rounded-md cursor-pointer grow"
        @change="performSort(($event.target as HTMLInputElement).value)"
      >
        <option v-for="opt in sortingOptions" :key="opt">{{ opt }}</option>
      </select>
    </div>
    <div class="flex items-center mt-4 md:mt-0 w-full md:w-auto">
      <input
        v-model="search"
        @input="performSearch(($event.target as HTMLInputElement).value)"
        placeholder="Search"
        class="rounded-md p-4 bg-website-background w-full border border-primary focus:border-primary"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref, defineEmits, defineProps } from "vue";

interface FilterBarProps {
  sortingOptions?: Array<string>;
}

defineProps<FilterBarProps>();
const emit = defineEmits(["search", "sort"]);

const search = ref("");

const performSearch = (val: string) => {
  emit("search", val.toLowerCase().trim());
};

const performSort = (selected: string) => {
  emit("sort", selected);
};
</script>
