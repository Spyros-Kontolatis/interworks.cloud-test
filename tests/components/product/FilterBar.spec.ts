import { shallowMount } from "@vue/test-utils";
import FilterBar from "../../../src/components/product/FilterBar.vue";

describe("FilterBar.vue", () => {
  it("renders the component", () => {
    const wrapper = shallowMount(FilterBar);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("shows sort options when property is passed", () => {
    const wrapper = shallowMount(FilterBar, {
      props: {
        sortingOptions: ["name-asc", "price-desc"],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.findAll("option")).toHaveLength(2);
  });
  it("correctly emits sort option when select is changed", async () => {
    const wrapper = shallowMount(FilterBar, {
      props: {
        sortingOptions: ["name-asc", "price-desc"],
      },
    });
    const select = wrapper.find("select");
    select.trigger("change");
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted().sort).toBeTruthy();
  });
  it("correctly emits search when input is changed", async () => {
    const wrapper = shallowMount(FilterBar);
    const input = wrapper.find("input");
    input.setValue("abc");
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted().search).toBeTruthy();
  });
});
