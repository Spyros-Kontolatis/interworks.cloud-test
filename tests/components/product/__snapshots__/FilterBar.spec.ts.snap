// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar.vue renders the component 1`] = `
<div
  class="flex flex-col md:flex-row w-full justify-between items-center"
>
  <!--v-if-->
  <div
    class="flex items-center mt-4 md:mt-0 w-full md:w-auto"
  >
    <input
      class="rounded-md p-4 bg-website-background w-full border border-primary focus:border-primary"
      placeholder="Search"
    />
  </div>
</div>
`;

exports[`FilterBar.vue shows sort options when property is passed 1`] = `
<div
  class="flex flex-col md:flex-row w-full justify-between items-center"
>
  <div
    class="flex gap-4 items-center w-full md:w-auto"
  >
    <h5
      class="font-extrabold text-start"
    >
      Sort:
    </h5>
    <select
      class="mt-2 p-4 bg-website-background border border-primary rounded-md cursor-pointer grow"
    >
      
      <option>
        name-asc
      </option>
      <option>
        price-desc
      </option>
      
    </select>
  </div>
  <div
    class="flex items-center mt-4 md:mt-0 w-full md:w-auto"
  >
    <input
      class="rounded-md p-4 bg-website-background w-full border border-primary focus:border-primary"
      placeholder="Search"
    />
  </div>
</div>
`;
